executor/src/witgen/affine_expression.rs

```diff
@@ -37,7 +39,7 @@ impl From<u32> for AffineExpression {
 impl AffineExpression {
     pub fn from_witness_poly_value(poly_id: usize) -> AffineExpression {
         AffineExpression {
-            coefficients: [vec![0.into(); poly_id], vec![1.into()]].concat(),
+            coefficients: BTreeMap::from([(poly_id, 1.into())]),
```

Comment:
what about the 0 that was in the vector?

Reply:
Previously, it would set `coefficients[poly_id] = 1` and insert enough zeros so that the vector is long enough. We don't need that anymore - that is actually the optimization here.